app/services/articles/detect_animated_images.rb

```diff
@@ -0,0 +1,38 @@
+module Articles
+  module DetectAnimatedImages
+    IMAGES_IN_LIQUID_TAGS_SELECTORS = [
+      ".liquid-comment img", # CommentTag
+      ".ltag-github-readme-tag img", # GithubReadmeTag
+      ".ltag__link__servicename img", # MediumTag
+      ".ltag__link__taglist img", # LinkTag
+      ".ltag__reddit--container img", # RedditTag
+      ".ltag__stackexchange--container img", # StackexchangeTag
+      ".ltag__twitter-tweet img", # TweetTag
+      ".ltag__user img", # UserTag and OrganizationTag
+      ".ltag__user-subscription-tag img", # UserSubscriptionTag
+      ".ltag_github-liquid-tag img", # GitHubIssueTag
+      ".podcastliquidtag img", # PodcastTag
+    ].join(", ").freeze
+
+    def self.call(article)
+      parsed_html = Nokogiri::HTML.fragment(article.processed_html)
+
+      # we ignore images contained in liquid tags as they are not animated
+      images = parsed_html.css("img") - parsed_html.css(IMAGES_IN_LIQUID_TAGS_SELECTORS)
+
+      found = false
+      images.each do |img|
+        src = img.attr("src")
+        next unless src
+
+        animated = FastImage.animated?(src)
```

Comment:
`FastImage` doesn't really raise exceptions unless asked, it will return `nil` (both if something went wrong or if the image is not animated) or `true`. See https://github.com/sdsykes/fastimage#features

I didn't specify a timeout because the default is 2 seconds and seems good enough. It's usually much faster:

```ruby
10.times.map { Benchmark.measure { FastImage.animated?("https://media1.tenor.com/images/38764a3868c2dd4f8c1253c4e9ec896e/tenor.gif") }.total }.min
# => 0.009350000000000025
```